src/bidiMapper/domains/network/NetworkManager.ts

```diff
@@ -155,14 +159,64 @@ export class NetworkManager {
       }
     );
 
+    // https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#event-requestPaused
     cdpTarget.cdpClient.on(
       'Fetch.requestPaused',
       (params: Protocol.Fetch.RequestPausedEvent) => {
         if (params.networkId) {
+          // The stage of the request can be determined by presence of
```

Comment:
nit: it's more readable if the event handler is extracted into a method and there is a clear place where the event handlers are removed.

Reply:
Done